substrate/frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -209,6 +210,8 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 				let return_type =
 					&call.methods.get(i).expect("def should be consistent with item").return_type;
 
+				remove_deprecation_attribute(&mut method.attrs);
```

Comment:
So this removes the deprecated attribute from the call?
I think it is not ideal, because we lose the deprecation of the call itself in rust code.

I think the ideal is to have the deprecation on the call function, also the deprecation on the call variant.
But then all usage of this call inside the macro should be used with `#[allow(deprecated)]` on top of the instruction. (We should be careful not to have this `#[allow(deprecated)]` for logic written by user).

Reply:
reverted this change

Reply:
I was thinking a more fine-grained implementation.

Like the `dispatch_bypass_filter` would be implemented with `#[allow(deprecated)]` just above the function call https://github.com/paritytech/polkadot-sdk/blob/74c7b9f3939ebfa6dc23f3516094af02b3af821f/substrate/frame/support/procedural/src/pallet/expand/call.rs#L460

Same for all other types: when the macro use them we would write: `#[allow(deprecated)]`.

But then we have to be careful not to write `#[allow(deprecated)]` on top of blocks which include code written by user. So that user is still notified when they use deprecated elements in their own code.

If it is too difficult, maybe we can merge this PR, without the revert then.

Reply:
I've tried implementing it this morning, and i can't seem to understand how to propagate `#[allow(deprecated)] from defs inside `#[pallet]` to expansion of `construct_runtime!`. Ie if we deprecate an event or an error we dont have any attributes to include `maybe_allow_attrs` inside code generated by `construct_runtime!`

Reply:
Yeah I agree here with @gui1117. Just removing the deprecation attribute, because it adds more noise is not the right way. The attribute is doing exactly for what it is designed for.

Reply:
> I've tried implementing it this morning, and i can't seem to understand how to propagate `#[allow(deprecated)]` from defs inside `#[pallet]` to expansion of `construct_runtime!`. Ie if we deprecate an event or an error we dont have any attributes to include `maybe_allow_attrs` inside code generated by `construct_runtime!`

about `construct_runtime!` if the whole call or event or pallet is deprecated, then I think the writer of the `construct_runtime!` should add `#[allow(deprecated)]` on top of the pallet variant:
```rust
construct_runtime!{
struct Runtime {
  #[allow(deprecated)]
  MyPalletWithDeprecatedCall: my_pallet; // my pallet is using call implicitly here.
}
}
```

Otherwise it is fair to actually warn the runtime writer than one of the pallet he uses has actual deprecated elements.

Note that this should only happen when deprecating whole types like `Event` or `Call` or `Pallet`, not when deprecating a single call inside the pallet `Call`.

Reply:
Not sure what the purpose of deprecating these types would be. I mean I would understand if the entire pallet would be deprecated.